test/integration/multicluster/install_test.go

```diff
@@ -165,63 +173,83 @@ func TestMulticlusterResourcesPostInstall(t *testing.T) {
 		{Namespace: "linkerd-multicluster", Name: "linkerd-gateway"},
 	}
 
-	for _, ctx := range contexts {
-		TestHelper.SwitchContext(ctx)
-		testutil.TestResourcesPostInstall(TestHelper.GetMulticlusterNamespace(), multiclusterSvcs, testutil.MulticlusterDeployReplicas, TestHelper, t)
-	}
+	TestHelper.SwitchContext(contexts[testutil.TargetContextKey])
+	testutil.TestResourcesPostInstall(TestHelper.GetMulticlusterNamespace(), multiclusterSvcs, testutil.MulticlusterDeployReplicas, TestHelper, t)
 }
 
 func TestLinkClusters(t *testing.T) {
-	linkName := "target"
-	// Get gateway IP from target cluster
-	lbCmd := []string{
-		"get", "svc",
-		"-n", "kube-system", "traefik",
-		"-o", "go-template={{ (index .status.loadBalancer.ingress 0).ip }}",
-	}
-	lbIP, err := TestHelper.KubectlWithContext("", contexts[testutil.TargetContextKey], lbCmd...)
+	linkCmds := make(map[string][]string)
+
+	// For each context construct the link command arguments
+	//  Source cluster will be linked against a target cluster without a gateway
+	//  Target cluster will be linked against a source cluster with a gateway and
+	//  with headless services
+	for k, ctx := range contexts {
```

Comment:
we're relying on the fact that contexts just contains the source and target contexts, right?  so wouldn't it be easier to just unroll this loop and run the link command for source and for target.  Instead of using a loop and an if?

Reply:
I thought about it, it would probably be easier to understand but more to read. I'll happily make the change since if it's making the steps clearer 👍🏻